bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -50,30 +67,187 @@ class DataTypeGroup(Enum):
-        """Initialize class with required attributes."""
-        self.name = name
-        self.data_type = data_type
-        self.column_type = column_type
-        self.status = status
-
-    def __eq__(self, other):
-        """Return attributes only if the referenced object is of type Column."""
-        if not isinstance(other, Column):
-            return NotImplemented
-        return (
-            self.name == other.name
-            and self.data_type == other.data_type
-            and self.column_type == other.column_type
-            and self.status == other.status
+    name: str
+    data_type: DataTypeGroup = attr.field(default=DataTypeGroup.UNDETERMINED)
+    column_type: ColumnType = attr.field(default=ColumnType.UNDETERMINED)
+    status: ColumnStatus = attr.field(default=ColumnStatus.UNDETERMINED)
+
+    """Validate the type of the attributes."""
+
+    @data_type.validator
+    def validate_data_type(self, attribute, value):
+        """Check that the type of data_type is as expected."""
+        if not isinstance(value, DataTypeGroup):
+            raise ValueError(f"Invalid {value} with type: {type(value)}.")
+
+    @column_type.validator
+    def validate_column_type(self, attribute, value):
+        """Check that the type of parameter column_type is as expected."""
+        if not isinstance(value, ColumnType):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+    @status.validator
+    def validate_status(self, attribute, value):
+        """Check that the type of parameter column_status is as expected."""
+        if not isinstance(value, ColumnStatus):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+
+@attr.define
+class Subset:
+    client: bigquery.Client()
+    destination_table: str = attr.field()
+    suffix: str = attr.field(converter=attr.converters.default_if_none(""))
+    query_cte: str = attr.field(converter=attr.converters.default_if_none(""))
+    dataset: str = attr.field(default=TEMP_DATASET)
+    project_id: str = attr.field(default=DEFAULT_PROJECT_ID)
+    expiration_days: Optional[float] = attr.field(default=None)
+
+    @destination_table.validator
+    def validate_table_version(self, attribute, value):
+        """Raise an exception if the version table name doesn't end with an integer."""
+        match = re.search(r"v(\d+)$", self.destination_table)
+        try:
+            version = int(match.group(1))
+            if not isinstance(version, int) or version <= 0:
+                raise click.ClickException(
+                    f"Destination table {value} must end with a positive version number."
+                )
+        except AttributeError:
+            f"Invalid table version for {value}"
+
+    @property
+    def expiration_ms(self) -> Optional[float]:
+        """Convert partition expiration from days to milliseconds."""
+        if self.expiration_days is None:
+            return None
+        return int(self.expiration_days * 86400000)
```

Comment:
```suggestion
        return int(self.expiration_days * 86_400_000)
```
tiny nit .. but makes it easier for me to see large numbers with splits ... can be ignored if you don't agree :)

Reply:
Done!